Assets/HoloToolkit/Utilities/Scripts/Performance/ViewportScaleManager.cs

```diff
@@ -0,0 +1,202 @@
+    /// reduction in pixels drawn.  Unlike the changing the render 
+    /// scale, this can be done after the holographic space is created 
+    /// and does not actually reduce the buffer size.
+    /// </summary>
+    public class ViewportScaleManager : Singleton<ViewportScaleManager>
+    {
+        public const int MainstreamVerticalResolution = 1280;
+
+        private float currentViewport = 1.0f;
+        public float CurrentViewport { get { return currentViewport; } }
+        private const float voiceControlledViewportDefault = -1;
+        private volatile float voiceControlledViewport = voiceControlledViewportDefault;
+
+        [HideInInspector]
+        public int MaxVirticalResolution = 0;
+        [HideInInspector]
+        public int MaxHorizontalResolution = 0;
+
+        public int CurrentVirticalResolution
+        {
+            get
+            {
+                return Mathf.RoundToInt(this.CurrentViewport * this.MaxVirticalResolution);
+            }
+        }
+        public int CurrentHorizontalResoution
+        {
+            get
+            {
+                return Mathf.RoundToInt(this.CurrentViewport * this.MaxHorizontalResolution);
+            }
+        }
+
+        [SerializeField]
+        private float minViewportScaleFactor = 0.05f;
+        [SerializeField]
+        private float maxViewportScaleFactor = 1.0f;
+
+        KeywordRecognizer keywordRecognizer = null;
+        Dictionary<string, System.Action> keywords = new Dictionary<string, System.Action>();
+
+        protected override void InitializeInternal()
+        {
+            var setViewport = XRSettings.renderViewportScale;
+            if (setViewport != 0)
+            {
+                currentViewport = setViewport;
+            }
+            else
+            {
+                currentViewport = 1.0f;
+            }
+
+            FindScreenResolution();
+
+            //Usefull for HoloLens, make sure to turn on mic capabilities in order to use it
+            //SetupKeywords();
+        }
+
+        public void SetupKeywords()
+        {
+            // Viewport kewords
+            for (int i = 1; i < 10; ++i)
+            {
+                float v = (float)i / 10.0f;
+                keywords.Add("Set viewport to point " + i, () =>
+                {
+                    voiceControlledViewport = v;
+                });
+
+                keywords.Add("Set viewport to point " + i + " five", () =>
+                {
+                    voiceControlledViewport = v + 0.05f;
+                });
+            }
+
+            keywords.Add("Set viewport to one", () =>
+            {
+                voiceControlledViewport = 1.0f;
+            });
+
+            keywords.Add("Set viewport to point zero five", () =>
+            {
+                voiceControlledViewport = 1.0f;
+            });
+
+            keywordRecognizer = new KeywordRecognizer(keywords.Keys.ToArray());
```

Comment:
We should probably use the ISpeechHandler instead of creating another instance of the keyword recognizer.

Reply:
Once again, I feel it is more useful to have fewer dependencies.  I would be fine with removing the voice commands, as they are simply an added utility for HoloLens.